robottelo/test.py

```diff
@@ -101,6 +102,99 @@
 LOGGER = logging.getLogger(__name__)
 
 
+class _AssertNotRaisesContext(object):
+    """A context manager used to implement :meth:`TestCase.assertNotRaises`.
+    """
+
+    def __init__(self, expected, test_case, expected_regex=None,
+                 cli_return_code=None, http_status_code=None):
```

Comment:
@ldjebran @abalakh usability is penalized but turns this in to more flexible solution, so I give +1 to @ldjebran suggestion.

If `value_handlers` are not passed we can have a `DefaultValueHandler` which automatically is `status_code` or `return_code` by inspecting if TestCase is a subclass of API or CLI, or maybe just raises an `AttributeError("value_handler is required")`